src/System.Management.Automation/engine/remoting/commands/EnterPSHostProcessCommand.cs

```diff
@@ -129,25 +140,34 @@ protected override void EndProcessing()
             }
 
             // Check selected process for existence, and whether it hosts PowerShell.
+            Runspace namedPipeRunspace = null;
             switch (ParameterSetName)
             {
                 case ProcessIdParameterSet:
                     Process = GetProcessById(Id);
+                    VerifyProcess(Process);
+                    namedPipeRunspace = CreateNamedPipeRunspace(Process.Id, AppDomainName);
```

Comment:
Interesting, do we really need the VerifyProcess() check (that is File.Exist())? Perhaps the runtime error is informative enough.

Reply:
I'm not quite sure what you're saying. This `VerifyProcess` is to keep the behavior consistent with the current behavior.